portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -59,19 +70,63 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the comment from the form.
-    String content = request.getParameter("title");
-    long timestamp = System.currentTimeMillis();
-    String ipAddr = request.getRemoteAddr();
+    final String content = request.getParameter("message");
+    final long timestamp = System.currentTimeMillis();
+    final String ipAddr = request.getRemoteAddr();
+
+    // Get the image from the form.
+    final String imageUrl = getUploadedFileUrl(request, "image");
 
     // Create entity for comment.
     Entity commentEntity = new Entity("Comment");
     commentEntity.setProperty("ip", ipAddr);
     commentEntity.setProperty("content", content);
     commentEntity.setProperty("timestamp", timestamp);
 
+    // If image submitted by user, add url to entity.
+    if (imageUrl != null) {
+      commentEntity.setProperty("imageUrl", imageUrl);
+    } else {
+      commentEntity.setProperty("imageUrl", "null");
```

Comment:
Why are we sending string "null" across the wire, it makes the client special case a specific string instead of inspecting to check if an attribute is set.

Reply:
I've changed this now to just straight up set imageUrl into the entity property.